### PR TITLE
Add segement replication upgrade changes for 2.8

### LINKED ISSUE #4001: [DOC] Update documentation for Segment Replication considerations for 2.8 release
**What do you want to do?**
 
- [x] Request a change to existing documentation
- [ ] Add new documentation
- [ ] Report a technical problem with the documentation
- [ ] Other 

**Tell us about your request.** Provide a summary of the request and all versions that are affected.

We are now including rolling upgrade support for the 2.8 release and need to see that reflected in the corresponding documentation: 
 Reference: https://opensearch.org/docs/latest/tuning-your-cluster/availability-and-recovery/segment-replication/index/#considerations 

We need to remove the 2nd bullet point in the above link that states rolling upgrades are not supported. 

**What other resources are available?** Provide links to related issues, POCs, steps for testing, etc.

### LINKED ISSUE #4003: [DOC] Add documentation for rolling upgrade for segment replication for 2.8 release
**What do you want to do?**
 
- [ ] Request a change to existing documentation
- [x] Add new documentation
- [ ] Report a technical problem with the documentation
- [ ] Other 

**Tell us about your request.** Provide a summary of the request and all versions that are affected.
 Add new documentation for segment replication rolling upgrade. 

Once we remove the rolling upgrade as a limitation for segment replication as specified in #4001, we need to add new documentation to specify how to perform rolling upgrades for segment replication. 

This could either be a new page or added as part of this: https://opensearch.org/docs/latest/install-and-configure/upgrade-opensearch/rolling-upgrade/ 

**What other resources are available?** Provide links to related issues, POCs, steps for testing, etc.

### PR DESCRIPTION
**Pull request by Naarcha-AWS**
- Fixes #4001 
- Fixes #4003 

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by kolchfa-aws** [comment]
@mch2 @anasalkouz Could you review this PR for technical accuracy and completeness please? Thanks!

File: _install-and-configure/upgrade-opensearch/index.md [hunk 1]
```diff
@@ -91,8 +91,11 @@ Upgrades spanning more than a single major version of OpenSearch will require ad
...
 
+[Segment replication]({{site.url}}{{site.baseurl}}/tuning-your-cluster/availability-and-recovery/segment-replication/index/) could help you continue to index data on your primary shard as you being a rolling upgrade especially if your cluster has a high index count and lower replica shard count. Enabling segment replication gives you the ability upgrade nodes without need to disable than reenable replica shards, allowing indexing to continue even as you perform a rolling upgrade. For instructions about how to enable segment replication, see [Segment replication configuration]({{site.url}}{{site.baseurl}}/tuning-your-cluster/availability-and-recovery/segment-replication/index/#segment-replication-configuration)
```
Comment (anasalkouz) [inline _install-and-configure/upgrade-opensearch/index.md:94]: Not sure if this is accurate. Enable segment replication, anyway need a full-reindex for the existing indexes. So, not sure if we can suggest users to use SegRep if they want to perform a rolling upgrade. I think we just remove the limitation and make sure users can do rolling upgrade while they are using SegRep.

Comment (mch2) [reply _install-and-configure/upgrade-opensearch/index.md:94]: +1

File: _install-and-configure/upgrade-opensearch/rolling-upgrade.md [hunk 2]
```diff
@@ -69,6 +69,19 @@ Review [Upgrading OpenSearch]({{site.url}}{{site.baseurl}}/upgrade-opensearch/in
      "transient" : { }
    }
    ```
+
+   If you have low replica count and prefer to keep shard replicas indexing data while you preform a rolling upgrade, you can enable [Segment replication](/tuning-your-cluster/availability-and-recovery/segment-replication/index/), as shown in the following example. This allows you to keep indexing data even as you take indexing nodes down for upgrades.
```
Comment (mch2) [inline _install-and-configure/upgrade-opensearch/rolling-upgrade.md:73]: same as above, they aren't able to enable segrep on an existing index.

**Comment by kolchfa-aws** [comment]
@anasalkouz Comments addressed. Could you review again and approve if there's nothing else to add? Thanks!

**Review by anasalkouz — APPROVED** [review]

**Review by cwillum — APPROVED** [review]

**Pull request merged**